core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -309,6 +309,52 @@ private Errors shareAcknowledgeError(String groupId, Uuid memberId, int reqEpoch
         }
     }
 
+    public List<TopicIdPartition> cachedTopicIdPartitionsInShareSession(String groupId, Uuid memberId) {
+        ShareSessionKey key = shareSessionKey(groupId, memberId);
+        ShareSession shareSession = cache.get(key);
+        if (shareSession == null) {
+            return null;
```

Comment:
Should it be null or an exception be thrown?

Reply:
agreed, it should be an exception